File: MangoActuator/src/network/http/ui/page.py
# -*- coding: utf-8 -*-
# @Project: 芒果测试平台
# @Description: 
# @Time   : 2024-09-03 16:25
# @Author : 毛鹏
from src.network.http.http_base import HttpBase


class Page(HttpBase):
    _url = '/ui/page'

    @classmethod
    def get_page(cls, page, page_size, params: dict = None):
        _params = {
            'page': page,
            'pageSize': page_size
        }
        if params:
            _params.update(params)
        return cls.get(Page._url, params=_params)

    @classmethod
    def post_page(cls, json_data: dict):
        return cls.post(Page._url, json=json_data)

    @classmethod
    def put_page(cls, json_data: dict):
        return cls.put(Page._url, json=json_data)

    @classmethod
    def delete_page(cls, _id, ):
        return cls.delete(Page._url, params={
            'id': _id,
        })

    @classmethod
    def module_page_name(cls, module_id, ):
        return cls.get(f'{Page._url}/name', params={
            'module_id': module_id,
        })

    @classmethod
    def page_copy(cls, page_id, ):
        return cls.post(f'{Page._url}/copy', json={
            'page_id': page_id,
        })

File: mango-console/src/api/uitest/page.ts
import { deleted, get, post, put } from '@/api/http'

export function getUiPage(data: object) {
  return get({
    url: '/ui/page',
    data: () => {
      return data
    },
  })
}

export function postUiPage(data: object) {
  return post({
    url: '/ui/page',
    data: () => {
      return data
    },
  })
}
export function putUiPage(data: object) {
  return put({
    url: '/ui/page',
    data: () => {
      return data
    },
  })
}
export function deleteUiPage(id: number | string[] | number[]) {
  return deleted({
    url: '/ui/page',
    data: () => {
      return {
        id: id,
      }
    },
  })
}

export function getUiPageName(moduleId: number) {
  return get({
    url: '/ui/page/name',
    data: () => {
      return {
        module_id: moduleId,
      }
    },
  })
}
export function postUiPageCopy(pageId: number | string) {
  return post({
    url: '/ui/page/copy',
    data: () => {
      return {
        page_id: pageId,
      }
    },
  })
}
